src/DotLiquid/FileSystems/ITemplateFileSystem.cs

```diff
@@ -0,0 +1,17 @@
+﻿namespace DotLiquid.FileSystems
+{
+    /// <summary>
+    /// This interface allow you return a Template instance,
+    /// it can reduce the template parsing time in some cases.
+    /// Please also provide the implementation of ReadTemplateFile for fallback purpose.
+    /// </summary>
+    public interface ITemplateFileSystem : IFileSystem
+    {
+        /// <summary>
+        /// Called by Liquid to retrieve a template instance
+        /// </summary>
+        /// <param name="templatePath"></param>
+        /// <returns></returns>
+        Template ReadTemplateInstance(Context context, string templateName);

```

Comment:
Since the purpose is to get the template, not always reading it, I think that `GetTemplate` is more appropriate.